tvaultuiv2/src/views/private/safe/components/Permissions/index.js

```diff
@@ -111,17 +113,39 @@ const useStyles = makeStyles(() => ({
 const Permissions = () => {
   const classes = useStyles();
   const [value, setValue] = useState(0);
-  const [users, setUser] = useState([]);
+  const [users, setUser] = useState({});
   const [addPermission, setAddPermission] = useState(false);
+  const [responseType, setResponseType] = useState(null);
+  const [toastMessage, setToastMessage] = useState('');
 
   const onSaveClicked = (data) => {
-    setUser((prev) => [...prev, data]);
+    setUser((prev) => ({ ...prev, [data.username]: data.access }));
     setAddPermission(false);
+    apiService
+      .postApiCall('/vault/v2/ss/sdb/user', data)
```

Comment:
```suggestion
      .postApiCall('/vault/v2/ss/sdb/user', data)
```

Should be part of apIservices.js

Reply:
fixed